src/com/twitter/intellij/pants/ui/CopyPathRelativeToBuildRootAction.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+package com.twitter.intellij.pants.ui;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.twitter.intellij.pants.util.PantsUtil;
+
+import java.awt.Toolkit;
+import java.awt.datatransfer.StringSelection;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+public class CopyPathRelativeToBuildRootAction extends AnAction {
+  @Override
+  public void actionPerformed(AnActionEvent event) {
+    Optional<VirtualFile> maybeBuildRoot = PantsUtil.findBuildRoot(event.getProject());
+    Optional<VirtualFile> maybeFile = PantsUtil.getFileForEvent(event);
+    if (!maybeBuildRoot.isPresent() || !maybeFile.isPresent()) {
```

Comment:
Could you DeMorgans this to `!(mBR.isPresent() && mF.isPresent())`. I think that'd be clearer and would mirror the expression used in `update`

Reply:
I've never seen DeMorgan used as a verb before, I like it :D Done!